libdune/entry.c

```diff
@@ -661,13 +661,17 @@ int dune_init(bool map_full)
 	}
 
 	setup_idt();
-
-	setup_apic();
+	if (!dune_setup_apic()) {
+		printf("dune: could not set up APIC\n");
+		ret = -ENOMEM;
+		goto err;
+	}
 
 	return 0;
 
 err:
 	// FIXME: need to free memory
```

Comment:
Do you still need the FIXME here?

Reply:
That was there originally... it must be a reference to something else